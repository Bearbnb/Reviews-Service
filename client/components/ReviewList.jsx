import React from 'react';
import ReactDOM from 'react-dom';
import Review from './Review.jsx';

let ReviewList = (props) => (
    <div className="review-list">
        {props.reviews.map(review =>
            <Review review={review}/>
        )}
    </div>
);

export default ReviewList;