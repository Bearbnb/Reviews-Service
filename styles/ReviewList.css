body {
    font-family: Circular,"Helvetica Neue",Helvetica,Arial,sans-serif;
    font-size: 14px;
    line-height: 1.43;
    color: #484848;
    background-color: #fff;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}
div {
    display: block;
}
.buttonContainer {
    display: inline-block;
    margin-bottom: 32px;
}

.circleButton{
    background-color: #008489 !important;
    color :#fff;
    border-radius: 100px
}
button {
    display: inline-block !important;
    background-color: transparent !important;
    border: none !important;
    color: #008489 !important;
    font: inherit !important;
    font-family: Circular,-apple-system,BlinkMacSystemFont,Roboto,Helvetica Neue,sans-serif !important;
    text-decoration: none !important;
    -webkit-appearance: button;
    cursor: pointer;
    -webkit-margin-before: 1em;
    -webkit-margin-after: 1em;
    -webkit-padding-start: 40px;
    text-transform: none;
    align-items: flex-start;
    text-align: center;
}