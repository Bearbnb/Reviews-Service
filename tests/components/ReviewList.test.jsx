import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import ReviewList from '../../client/components/ReviewList';

describe('<ReviewList />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<ReviewList />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('should call HandlePrev when clicking < button', () => {
  //   const handle = jest.fn();
  //   const wrapper = shallow(<ReviewList
  //     currentPage={5}
  //     reviewlist={[{}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}]}
  //     pages={1, 2, 3, 4, 5, 6},
  //     HandlePrev = { handle}
  //   />);
  //   wrapper.find('div > div > div > button').first().simulate('click');
  //   expect(HandlePrev).toHaveBeenCalled();
  // });
});
