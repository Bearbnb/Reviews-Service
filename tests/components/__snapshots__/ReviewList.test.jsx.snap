// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> should render properly 1`] = `
<div
  className="reviewlist"
>
  <FlagModal
    hideModalFunc={[Function]}
    showModal={false}
  />
  <div
    className="reviewlist"
  />
  <div>
    <div
      className="buttonContainer"
    >
      <button
        id="nextButton"
        onClick={[Function]}
        type="button"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;
